wallet-new/src/Cardano/Wallet/Kernel/PrefilterTx.hs

```diff
@@ -295,6 +307,17 @@ prefilterBlock block wid esk =
 
     accountIds = Map.keysSet inpAll `Set.union` Map.keysSet outAll
 
+    toWalletKey :: (WalletId, EncryptedSecretKey) -> WalletKey
+    toWalletKey (wid, esk) = (wid, keyToWalletDecrCredentials $ KeyForRegular esk)
+
+-- Prefilter a block for a single wallet
+prefilterBlock :: ResolvedBlock
+               -> WalletId
+               -> EncryptedSecretKey
+               -> (Map HdAccountId PrefilteredBlock, [TxMeta])
+prefilterBlock block wid esk =
+    prefilterBlockForWallets block [(wid, esk)]
```

Comment:
I don't think this is useful provided we have the other one; especially because it'll open the gate for doing the same as before for prefiltering. I'd rather not have `prefilterBlockForWallets` but simply name it `prefilterBlock` as it was before, changing it's signature to accept a list instead of a single key.

Reply:
we do still need `prefilterBlock` for a single wallet during restoration, but I hear you, that can just call the unified one with a single wallet key